File: frontend/src/app/models/producto.ts
import { Categoria } from './categoria';
import { Dimension } from './dimension';
import { Detalle_factura } from './detalle_factura';

export class Producto{
    id_producto: number; 
    cod_producto: string;
    nom_producto: string; 
    descripcion: string; 
    costo_compra: number;
    precio_unitario: number; 
    cantidad_existente: number; 
    stock_minimo: number; 
    stock_maximo: number;
    id_categoria: number; 
    categoria: Categoria;
    habilitado: number;
    //detalles_kardex: Array<Detalle_kardex>;
    detalles_factura: Array<Detalle_factura>;
    dimensiones: Array<Dimension>;
 }

File: backend/Models/Producto.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Collections.Generic;

namespace backend.Models
{
    [Table("tb_producto")]
    public class Producto
    {
        [Key]
        public int id_producto { get; set; }
        [Column(TypeName = "varchar(45)")]
        public string cod_producto { get; set; }
        [Column(TypeName = "varchar(60)")]
        public string nom_producto {get;set;}
        [Column(TypeName = "varchar(150)")]
        public string descripcion {get;set;}
        public decimal costo_compra {get;set;}
        public decimal precio_unitario{get;set;}
        public int cantidad_existente {get;set;}

        public int habilitado { get; set; }
        [ForeignKey("id_categoria")]
        public int id_categoria{get;set;}
        public Categoria categoria {get;set;}
        public List<Detalle_kardex> detalles_kardex{get;set;}
        public List<Detalle_factura> detalles_factura{get;set;}
        public List<Dimension> dimensiones{get;set;}

        
    }
}